### DISCUSSION TITLE
Deno.run(...) in test

### DISCUSSION THREAD

**Discussion by tamusjroyce** [denoland/deno#11606] [Q&A] [answered]
I have a deno test which executes version. Pretty simple. And deno even detects that this is leaking resources!!!

> `deno test --allow-run`

deno.test.ts
```
Deno.test("deno runs", async () => {
    const p = Deno.run({
        cmd: ["deno", "--version"],
        stdout: "piped",
        stderr: "piped",
    });
    console.log(JSON.stringify(await p) + '');
    const output = await p.output();
    const outStr = new TextDecoder().decode(output);
    // const err = await p.error();
    // const errStr = new TextDecoder().decode(err);
    p.close();
});
```
**Output:**
deno runs
AssertionError: Test case is leaking resources.
  "2": "stderr"
}
After: {
  "0": "stdin",
  "1": "stdout",
  "2": "stderr",
  "4": "childStderr"
}

vs
```
Deno.test("deno runs", async () => {
    const p = Deno.run({
        cmd: ["deno", "--version"],
        stdout: "piped",
        stderr: "piped",
    });
    console.log(JSON.stringify(await p) + '');
    // const output = await p.output();
    // const outStr = new TextDecoder().decode(output);
    // const err = await p.error();
    // const errStr = new TextDecoder().decode(err);
    p.close();
});
```
Before: {
  "0": "stdin",
  "1": "stdout",
  "2": "stderr"
}
After: {
  "0": "stdin",
  "1": "stdout",
  "2": "stderr",
  "4": "childStderr"
}

Should awaiting on output() be required to clean up resources? How do I await on p.error()? error isn't a function

**Comment by lucacasonato** [comment, marked as answer]
await p.stderrOutput()

**Reply by tamusjroyce** [reply]
I should have dug harder. Now I can get my much improved unit test working. Thank you!
